posts/2019-01-14-easy-peasy-nix-versions.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Easy Peasy Nix Versions
+---
+
+<style>
+.story {
+  background-color: lightblue;
+  padding: 20px;
+  border-radius: 10px;
+}
+
+</style>
+
+# Easy Peasy Nix Versions
+
+This is a convention for using third-party packages in Nix. It has a simple
+directory structure, makes using the packages straightforward and automatizes
+updates.
+
+---
+
+Where should you declare the third-party packages you use in your Nix project?
+Should all specs be declared in the `default.nix`? Should each third-party
+package get its own directory with a `default.nix` and a `spec.src.json`? Or
+maybe the specs should only exist at call site, potentially duplicated if the
+package is used in different places? I settled on a simple convention which
+I've been using for a year across all my personal and work Nix projects.
+
+The idea is to use a single JSON file to store all versions rather than having
+each package in a directory defining its URL and version separately (I tend to
+get lost with the latter). Two more files are involved: one that acts as glue
+between JSON and Nix, and one that automatizes the package updates. It's not a
+full-blown solution like [require.nix] but it does the job very well for small-
+to medium-sized project.
```

Comment:
Who is the intended audience?

I see exactly what the problem is, because I managed nixified repos before. For a more newbie audience it might be worth being more explicit about the issue by showing an example of what not to do.